thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/dashboard/resources/AnomalyFlattenResource.java

```diff
@@ -0,0 +1,119 @@
+import javax.ws.rs.Path;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.pinot.thirdeye.anomalydetection.context.AnomalyFeedback;
+import org.apache.pinot.thirdeye.common.dimension.DimensionMap;
+import org.apache.pinot.thirdeye.datalayer.bao.MergedAnomalyResultManager;
+import org.apache.pinot.thirdeye.datalayer.dto.MergedAnomalyResultDTO;
+import org.apache.pinot.thirdeye.datasource.DAORegistry;
+
+
+@Path("thirdeye/table")
+public class AnomalyFlattenResource {
+  private static final DAORegistry DAO_REGISTRY = DAORegistry.getInstance();
+  private MergedAnomalyResultManager mergedAnomalyResultDAO;
+  public static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  public static final String ANOMALY_ID = "anomalyId";
+  public static final String ANOMALY_COMMENT = "comment";
+  public static final String FUNCTION_ID = "functionId";
+  public static final String WINDOW_START = "start";
+  public static final String WINDOW_END = "end";
+  public static final String DIMENSION_KEYS  = "keys";
+
+  public static final List<String> REQUIRED_JSON_KEYS =Arrays.asList(FUNCTION_ID, WINDOW_START, WINDOW_END);
+  public static final String DEFAULT_DELIMINATOR = ",";
+
+  public AnomalyFlattenResource() {
+    this.mergedAnomalyResultDAO = DAO_REGISTRY.getMergedAnomalyResultDAO();
+  }
+
+  public AnomalyFlattenResource(MergedAnomalyResultManager mergedAnomalyResultDAO) {
+    this.mergedAnomalyResultDAO = mergedAnomalyResultDAO;
+  }
+
+  /**
+   * Flatten a list of anomaly results to a list of map
+   * @param jsonPayload a json string
+   * @return a list of map
+   * @throws IOException
+   */
+  @GET
+  @ApiOperation(value = "View a flatted merged anomalies for collection")
+  public List<Map<String, String>> flatAnomalyResults(String jsonPayload) throws IOException {
+    if (StringUtils.isBlank(jsonPayload)) {
+      throw new IllegalArgumentException("jsonPayload cannot be null or empty");
+    }
+    Map<String, Object> inputMap = OBJECT_MAPPER.readValue(jsonPayload, Map.class);
+
+    // Assert if reuired keys are in the map
+    for (String requiredKey : REQUIRED_JSON_KEYS) {
+      if (!inputMap.containsKey(requiredKey)) {
+        throw new IllegalArgumentException(String.format("Miss %s in input json String; %s are required", requiredKey,
+            REQUIRED_JSON_KEYS.toString()));
+      }
+    }
+
+    // Retrieve anomalies
+    long functionId = Long.valueOf(inputMap.get(FUNCTION_ID).toString());
+    long start = Long.valueOf(inputMap.get(WINDOW_START).toString());
+    long end = Long.valueOf(inputMap.get(WINDOW_END).toString());
+    List<MergedAnomalyResultDTO> anomalies = mergedAnomalyResultDAO.
+        findByStartTimeInRangeAndDetectionConfigId(start, end, functionId);
+
+    // flatten anomaly result information
+    List<String> tableKeys = null;
+    if (inputMap.containsKey(DIMENSION_KEYS)){
+      Object dimensionKeysObj = inputMap.get(DIMENSION_KEYS);
+      if (dimensionKeysObj instanceof List) {
```

Comment:
Why there could be two different types?

Reply:
The dimension key should be in list; for corner cases, the dimension key is not parsed as list, the second condition can make up the problem.